### ISSUE TITLE
NPE in VariableListView 

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#12987]
### Describe your problem!

Found on GooglePlayConsole for 2022.05.02-RC. Not sure whether already solved by #12980 or not.

```
java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at cgeo.geocaching.utils.formulas.VariableMap.recalculate (VariableMap.java:300)
  at cgeo.geocaching.utils.formulas.VariableMap.remove (VariableMap.java:158)
  at cgeo.geocaching.utils.formulas.VariableList.removeVariable (VariableList.java:158)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.removeVarAt (VariableListView.java:380)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.lambda$onCreateViewHolder$5 (VariableListView.java:296)
  at cgeo.geocaching.ui.VariableListView$VariablesListAdapter.lambda$onCreateViewHolder$5$VariableListView$VariablesListAdapter (Unknown Source)
  at cgeo.geocaching.ui.-$$Lambda$VariableListView$VariablesListAdapter$FAGdGRunLlpx8rbyjIVxoohm8Zo.onClick (Unknown Source:4)
  at android.view.View.performClick (View.java:7455)
  at com.google.android.material.button.MaterialButton.performClick (MaterialButton.java:1119)
  at android.view.View.performClickInternal (View.java:7432)
  at android.view.View.access$3700 (View.java:835)
  at android.view.View$PerformClick.run (View.java:28810)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:201)
  at android.os.Looper.loop (Looper.java:288)
  at android.app.ActivityThread.main (ActivityThread.java:7870)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:548)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1003)
```

### How to reproduce?

unclear

### Reproducible

Unclear

### c:geo Version

2022.05.02-RC

**Comment by Lineflyer** [comment]
Another one to double check:

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:5160)
  at android.app.ActivityThread.performPauseActivity (ActivityThread.java:5111)
  at android.app.ActivityThread.handlePauseActivity (ActivityThread.java:5063)
  at android.app.servertransaction.PauseActivityItem.execute (PauseActivityItem.java:47)
  at android.app.servertransaction.ActivityTransactionItem.execute (ActivityTransactionItem.java:45)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2260)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:210)
  at android.os.Looper.loop (Looper.java:299)
  at android.app.ActivityThread.main (ActivityThread.java:8166)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:556)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1037)
Caused by: java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at cgeo.geocaching.utils.formulas.VariableList.getEntries (VariableList.java:237)
  at cgeo.geocaching.models.CacheVariableList.saveState (CacheVariableList.java:70)
  at cgeo.geocaching.VariablesViewPageFragment.checkUnsavedChanges (VariablesViewPageFragment.java:124)
  at cgeo.geocaching.VariablesViewPageFragment.onPause (VariablesViewPageFragment.java:158)
  at androidx.fragment.app.Fragment.performPause (Fragment.java:3168)
  at androidx.fragment.app.FragmentStateManager.pause (FragmentStateManager.java:632)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:314)
  at androidx.fragment.app.FragmentStore.moveToExpectedState (FragmentStore.java:112)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1647)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:3128)
  at androidx.fragment.app.FragmentManager.dispatchPause (FragmentManager.java:3090)
  at androidx.fragment.app.FragmentController.dispatchPause (FragmentController.java:284)
  at androidx.fragment.app.FragmentActivity.onPause (FragmentActivity.java:390)
  at cgeo.geocaching.activity.AbstractActivity.onPause (AbstractActivity.java:118)
  at cgeo.geocaching.CacheDetailActivity.onPause (CacheDetailActivity.java:443)
  at android.app.Activity.performPause (Activity.java:8389)
  at android.app.Instrumentation.callActivityOnPause (Instrumentation.java:1531)
  at android.app.ActivityThread.performPauseActivityIfNeeded (ActivityThread.java:5150)
```

**Comment by eddiemuc** [comment]
Given current code formatting state: when can I start fixing this problem in release branch?

**Comment by Lineflyer** [comment]
The code formatting commits are already contained on `release` as branches have been merged (again).

So, there is AFAICS no restriction in providing PRs towards `release` and later merging upstream to `master`.

**Comment by Lineflyer** [comment]
Another one for the new beta version 2022.05.09-RC:

```
java.lang.NullPointerException: 
  at java.util.Objects.requireNonNull (Objects.java:220)
  at cgeo.geocaching.utils.formulas.VariableList.getEntries (VariableList.java:240)
  at cgeo.geocaching.models.CacheVariableList.saveState (CacheVariableList.java:70)
  at cgeo.geocaching.VariablesViewPageFragment.lambda$checkUnsavedChanges$10 (VariablesViewPageFragment.java:126)
  at cgeo.geocaching.VariablesViewPageFragment.lambda$checkUnsavedChanges$10$VariablesViewPageFragment (Unknown Source)
  at cgeo.geocaching.-$$Lambda$VariablesViewPageFragment$9xrHxJncNtZbSJWlQLo_BWhT4KA.run (Unknown Source:2)
  at android.os.Handler.handleCallback (Unknown Source:2)
  at android.os.Handler.dispatchMessage (Unknown Source:4)
  at android.os.Looper.loop (Unknown Source:249)
  at android.app.ActivityThread.main (Unknown Source:134)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (Unknown Source:11)
  at com.android.internal.os.ZygoteInit.main (Unknown Source:313)
```

**Comment by eddiemuc** [comment]
Hopefully fixed with PR #12997
